core/src/apps/management/recovery_device/layout.py

```diff
@@ -48,12 +48,38 @@ async def request_mnemonic(
 
     await button_request("mnemonic", code=ButtonRequestType.MnemonicInput)
 
+    # Allowing to go back to previous words, therefore cannot use just loop over range(word_count)
     words: list[str] = []
-    for i in range(word_count):
+    i = 0
+    while True:
+        # All the words have been entered
+        if i >= word_count:
+            break
+
+        # Not allowing negative indexes
+        if i < 0:
```

Comment:
...and this should be folded into `if not word` below so that we never start with a negative index in the first place

Reply:
Done in [33ce5da](https://github.com/trezor/trezor-firmware/pull/3458/commits/33ce5da629fc0aeb862c95a9ebe04027ef058fa2)